docs/source/training_agents.md

```diff
@@ -0,0 +1,138 @@
+# Training Agents with Environments in TRL
+
+Environments in TRL provide a powerful way to customize the rollout process for reinforcement learning (RL) with language models, specifically when using vLLM as the backend. This enables advanced agent training workflows, such as allowing your model to interact with external tools or APIs during RL training.
```

Comment:
nit since we emphasise the backend in the note below
```suggestion
Environments in TRL provide a powerful way to customize the rollout process for reinforcement learning (RL) with language models. This enables advanced agent training workflows, such as allowing your model to interact with external tools or APIs during RL training.
```